extends Node


# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼
# å…±é€šãƒ¡ãƒ¢ãƒªé–¢é€£
# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼

# å…ˆç¥–ã®è¾æ›¸ã‚­ãƒ£ãƒƒã‚·ãƒ¥
var ancestors = {}


# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼
# å¤–ãƒ‘ã‚¹é–¢é€£
# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼


# ãƒ—ãƒ­ã‚°ãƒ©ãƒ ã‚ºãƒ»ãƒãƒ–å–å¾—
func hub():
	return MonkeyHelper.find_ancestor_child(
			self,
			"ğŸŒProgrammer/ğŸ›©ï¸Hub",
			self.ancestors)


# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼
# ä»¥ä¸‹ã€ä¸»è¦ãƒ—ãƒ­ã‚°ãƒ©ãƒ 
# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼


# Called when the node enters the scene tree for the first time.
func _ready():
	pass # Replace with function body.


# Called every frame. 'delta' is the elapsed time since the previous frame.
func _process(delta):
	pass


# TODO ã“ã“ã«ã‚·ãƒŠãƒªã‚ªã‚’èª­å–ã‚‹å‡¦ç†ã‚’ç§»å‹•ã—ãŸã„
func parse_paragraph_test():
	print("ï¼»ã‚·ãƒŠãƒªã‚ªã‚¨ãƒ³ã‚¸ãƒ³ï¼½ã€€æº–å‚™ä¸­")
	pass


# é€šå¸¸æ–‡æ›¸ã®è¡¨ç¤º	
func print_normal_text(paragraph_text):
	#print("ï¼»ã‚·ãƒŠãƒªã‚ªã‚¨ãƒ³ã‚¸ãƒ³ï¼½ã€€æº–å‚™ä¸­ã€€é€šå¸¸æ–‡æ›¸ã®è¡¨ç¤º")
	self.hub().get_instruction(&"ğŸ“˜NormalText").do_it(paragraph_text)


# é¸æŠè‚¢ã®è¡¨ç¤º
func print_choices(paragraph_text):
	print("ï¼»ã‚·ãƒŠãƒªã‚ªã‚¨ãƒ³ã‚¸ãƒ³ï¼½ã€€æº–å‚™ä¸­ã€€é¸æŠè‚¢ã®è¡¨ç¤º")
	var message_window_gui = self.hub().get_current_message_window_gui()

	# é¸æŠè‚¢ã 
	if message_window_gui.choices_row_numbers != null:
		self.hub().get_instruction(&"ğŸ“˜NormalTextChoice").do_it(paragraph_text)
		return true

	return false
